### ISSUE TITLE
Samples & VSIX

### ISSUE THREAD

**Issue by samcov** [dotnet/orleans#54]
The source code builds properly, however, when I run "InstallOrleansVSTools.cmd", it fails to find what it needs or there is a permission problem is what it reports(see pic).

Since that failed, I installed the VSIX manually, but the Hello World sample fails to build.

I originally tried this in VS2012, but got the same results in VS2013 on a different machine.
![untitled](https://cloud.githubusercontent.com/assets/1551314/5970561/af4b74dc-a7e9-11e4-8ec1-5b7f7ec7ec69.jpg)

**Comment by jkonecki** [comment]
I'll take a look at it as I'm working on the samples right now.

**Comment by jthelin-microsoft** [comment]
This is most likely a problem with the slightly changed directory structure in the SDK installer after the switch from CodePlex to GitHub..

**Comment by sergeybykov** [comment]
The GitHub version of the msi doesn't include InstallOrleansVSTools.cmd, only UninstallOrleansVSTools.cmd. The msi registers the VSIX during installation. You can always double-click on the VSIX to install it. What happens when you do that?

**Comment by samcov** [comment]
@sergeybykov - Thanks for you attempt to help, but as I stated in the original post, I do have InstallOrleansVSTools.cmd, and I did manually install the VSIX after it failed, and it installs. I simply reported that in order to help others and of course get it corrected for the future.

However, the code does not work for the simplest case(HelloWorld).  IMO, there's just something simple to correct, since I know others that didn't have these problems with the Codeplex bits.

@jthelin-microsoft - Yes, I absolutely agree that is likely the problem with the VSIX install

**Comment by sergeybykov** [comment]
@samcov - I think there are three different issues here.

1. InstallOrleansVSTools.cmd is included in the sources while it's not part of the msi anymore, and is not really needed. It is supposed to be run from the installed SDK folder (C:\Microsoft Project Orleans SDK v1.0\SDK) and it assumes the old directory structure that's changed. We should probably simply delete it.

2. VSIX gets installed by the msi, and re-installing of it, e.g. by double-clicking, will work only after you uninstall it first (with the included UninstallOrleansVSTools.cmd.

3. The msi registers an environment variable OrleansSDK which points to the installation folder. Projects created with the VSIX templates depend on it, and won't build without it. I think this is likely why you are unable to build HelloWorld. There is a known issue with OrleansSDK being created but sometimes not propagated through the system with a workaround described in https://github.com/dotnet/orleans/wiki/Installation. I think the msi lacks a custom action to force propagation of the environment variable. Can you check if the environment variable workaround will make HelloWorld build for you?

**Comment by samcov** [comment]
@sergeybykov - There is a lot of references to installing the msi, but I've never seen where any msi is located. It's claimed that you can get it on codeplex, but the link on that page comes here, and there is no msi.

BTW, you were correct about the environment variable, it obviously doesn't exist since I have never used the msi.

**Comment by gabikliot** [comment]
http://dotnet-ci.cloudapp.net/job/dotnet_orleans/ws/Binaries/Release/orleans_setup.msi

**Comment by samcov** [comment]
Got it!

**Comment by samcov** [comment]
Ok, VERY good news, everything compiled after I "Edited" the environment variables!

A little tough, but Orleans is worth it :)
